### ISSUE TITLE
[BUG] Unable to skip feature importance calculation in tests

### ISSUE THREAD

**Issue by DL1992** [deepchecks/deepchecks#1513]
**Describe the bug**
Running tests that involve calculating feature importance by deepchecks crashes. trying to skip these calculations in test suits by using feature_importnace_timeout=0 or/and feature_importance_force_permutation=False does not work. the tests still try to calculate FI and than the notebook crash.
**To Reproduce**
Sadly, I cant share the dataset I'm using or the code.
run test suit with feature_importnace_timeout=0 and a model that has no feature importance - feature importance is still calculating

**Expected behavior**
No test should try to calculate feature importance - those that must have FI do should be skipped

**Environment (please complete the following information):**
 - OS: linux
 - Python Version: 3.7.3
 - Deepchecks Version:0.6.4

**Additional context**

**Comment by noamzbr** [comment]
Hi @DL1992, thanks for reporting! 
I think I know what's going on. The WholeDatasetDrift and ModelErrorAnalysis checks calculate their own (internal) feature importance for their own (internal) models, and this internal calculation also raises the "still calculating" text.

**To confirm if this is the case**
1. Do you also see a text like the following when limiting the timeout?
![image](https://user-images.githubusercontent.com/17730502/170878413-276e04a9-30c0-4e29-8b9e-f24a0b0fb46c.png)
2. How long are the calculations that do happen? Should be very few seconds. 
3. To make sure - what suite where you using?
4. We can also make sure by running individual checks that need feature importance, such as TrainTestFeatureDrift, and checking if limiting the timeout works for them.

**What should we do**
Given that my assumption is correct (pending on the input from @DL1992), we should make sure that there is no such printout from internal FI calculations on internal models that are not the user model.

**Comment by DL1992** [comment]
Hi.
No, I see text that says "calculating feature importance" or something like this.
the notebook crash after a few second - maybe a minute of trying to calculate FI

it happens in model evaluation suit (unused feature test) or validation suit when using whole dataset drift

**Comment by noamzbr** [comment]
OK that's odd. 
We should still do what I outlined earlier (not print FI calc for internal models), but that won't solve the issue.

We need to try and reproduce this issue. @DL1992 if by any chance you can attach here a reproducible example that causes that print (even with one of the built-in deepchecks dataset) that could be great!

**Comment by DL1992** [comment]
Hi, can confirm the problem still exist in 0.7.2.
It crashes all check that involves FI. also crashes scripts with error 139 Linux segmentation error
![1654716717858](https://user-images.githubusercontent.com/21197955/172701861-0079c453-8e5b-4989-bdd4-28df5c29916b.jpeg)

**Comment by JKL98ISR** [comment]
looks like it does skip but still prints the warning, should add a case for 0.
the crashing bug is in another issue too, so it's just to ignore the message
